tests/python_package_test/test_sklearn.py

```diff
@@ -1561,6 +1584,27 @@ def test_validate_features(task):
     model.predict(df2, validate_features=False)
 
 
+# LightGBM's 'predict_disable_shape_check' mechanism is intentionally not respected by
+# its scikit-learn estimators, for consistency with scikit-learn's own behavior.
+@pytest.mark.parametrize("predict_disable_shape_check", [True, False])
+def test_predict_rejects_inputs_with_incorrect_number_of_features(predict_disable_shape_check):
+    X, y, _ = _create_data(task="regression", n_features=4)
```

Comment:
I think it's better to use classification here and test `predict_proba()` as well.

Reply:
great point, I agree. Thinking about it more, I think there's good reason to just parametrize this over all tasks, since each is slightly different (e.g. classification has 2 `predict()` and `predict_proba()`, ranking inherits directly from `LGBMModel` with no mixin).

Reply:
I've updated this test to use classification, regression, and ranking... and to call both `predict()` and `predict_proba()`. Let me know if you think it looks too complicated with all the `if` statements and I'll happily change it to just classification.

Reply:
I support your choice with parametrization!